### PR TITLE
Docs/use kedro sphinx theme

### LINKED ISSUE #1882: Enhance Kedro-viz doc styling by using kedro-sphinx-theme
## Description
To follow what @astrojuanlu has changed for Kedro doc, Kedro-viz doc also should use [kedro-sphinx-theme](https://github.com/kedro-org/kedro-sphinx-theme) for consistency

## Possible Implementation
Possible implementation can be similar to [3675](https://github.com/kedro-org/kedro/pull/3675)

Task includes:

- [ ] Update `html_theme = "kedro-sphinx-theme"` instead of `html_theme = "sphinx_rtd_theme"`
- [ ] Double check if some of the styling functions in `docs/source/conf.py` are already defined in kedro-sphinx-theme repo? If not, can we move them there to be able to reuse them?
- [ ] Update kedro-sphinx-theme packages where needed

## Checklist

- [ ] Include labels so that we can categorise your feature request

### PR DESCRIPTION
**Pull request by Huongg**
## Description

Fixes #1882 

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by astrojuanlu** [comment]
Hmmm don't know why, but instead of picking up `kedro-sphinx-theme`, the doc looks like `sphinx-rtd-theme` without our styling!

https://kedro--1898.org.readthedocs.build/projects/kedro-viz/en/1898/

File: demo-project/docs/source/conf.py [hunk 1]
```diff
@@ -92,7 +92,7 @@
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
 #
-html_theme = "sphinx_rtd_theme"
+html_theme = "kedro-sphinx-theme"
```
Comment (rashidakanchwala) [inline demo-project/docs/source/conf.py:95]: we need to change this here  kedro-viz/docs/source/conf.py

Comment (astrojuanlu) [reply demo-project/docs/source/conf.py:95]: 🦅 good catch!

Comment (Huongg) [reply demo-project/docs/source/conf.py:95]: ah i see thanks @rashidakanchwala

**Comment by astrojuanlu** [comment]
Now that looks OK https://kedro--1898.org.readthedocs.build/projects/kedro-viz/en/1898/ ✔️

File: demo-project/docs/source/conf.py [hunk 2]
```diff
@@ -92,7 +92,7 @@
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
 #
-html_theme = "sphinx_rtd_theme"
```
[the remark this answers is not in the document]
Comment (astrojuanlu) [reply demo-project/docs/source/conf.py:95]: Good call @ankatiyar , let's keep consistency with our starters for now.

[the remark this answers is not in the document]
Comment (Huongg) [reply demo-project/docs/source/conf.py]: done, thanks @ankatiyar

**Review by ankatiyar — APPROVED** [review]
Thanks @Huongg! 🚀

**Pull request merged**